### ISSUE TITLE
Need some advice

### ISSUE THREAD

**Issue by gbillus** [Maaack/Godot-Game-Template#156]
The menu template is honestly really awesome. I can mostly use it without any modification. I changed a menu or two and I was able to get it to flow correctly. I can heavily reuse this by extending existing functionality. There are only 2 changes I can think. I dont know if these would be a good fit for the project or not but I certainly need this on my copy. 

1 I need the loading screen to not close immediately. I want to play a cutscene in the background that has crucial info about the next match in the first 30 seconds. I dont want it to be skippable until it reaches a certain point(would probably just wire with a timer for now). Im gonna write it so when the 30 second mark is reached and level is loaded, a play button appears so I can confirm the player has a hand on the controller/mouse. the play button would finish the level transition. I tried poking thru the code but I kept tracing function calls with no clear understanding of the structure. Quite frankly, I dont understand where level loading, or loading screen hiding, occurs.

2 There is an awesome plugin called Multiplay-Core that helps abstract some of the complexities of handling multiple players. https://github.com/maji-git/multiplay-core
It offers abstraction for the network pipe, input handling, player/scene loading and sync, ectr. You can probably see what I need already. Most of that is either unrelated or trivial to switchover. The thing that  I need help understanding there is how to load my scene. I want to reuse the menus and as much of the logic as possible, as I recognize that it is better than what I have come up with in the past.

I realize you have your own things to deal with, but any thoughts that you might have, even loose ramblings, would be greatly appreciated. Thank you again for you time and this project.

**Comment by Maaack** [comment]
Thanks for all of the feedback!

1. The loading screen is a little hard to follow and it currently can be used in two ways. Either as the only scene in the scene tree (when calling `SceneLoader.load_scene(...)`), or as a background node in the current scene (when calling `$LoadingScreen.reset()`). In the first case, it'll get replaced by the next scene with a call to `SceneLoader.change_scene_to_resource()`. In the second case, the node gets hidden by a call to `$LoadingScreen.hide()` in `game_ui.gd` and the level is placed on the node above it. I've successfully delayed the closing in another project that waited for audio to finish playing during the loading screen, so it can be done. Unfortunately, the project is private, so I can't share it directly here.
2. In `game_ui.tscn`, you can either have the [LevelLoader](https://github.com/Maaack/Godot-Game-Template/blob/main/addons/maaacks_game_template/docs/GameSceneSetup.md#level-loading) load your level, or drop your scene as a node into the `SubViewport`. Alternatively, you can load your scene by opening `main_menu_with_animations.tscn`, selecting the root node, and setting the `game_scene_path` parameter to the path of your scene. However, you'll lose the minor benefits of the included `game_ui.tscn`.

Hope that helps and good luck!

**Comment by Maaack** [comment]
Closing this issue for now. There might be some work to do in the future to make the LoadingScreen easier to understand and customize.
